frontend/templates/troubleshooting/server.tsx

```diff
@@ -53,25 +54,37 @@ export const getServerSideProps: GetServerSideProps<
                      ifixitOrigin,
                   },
                   { forceMiss: hasDisableCacheGets(context) }
-               )
-         )
+               ).catch(() => null)
+            );
+         })
       );
       return {
          ...solution,
-         guides,
+         guides: guides.filter((guide): guide is Guide => Boolean(guide)),
          products: products.filter((product): product is ProductType =>
             Boolean(product)
          ),
       };
    }
 
-   const troubleshootingData = await client.get<TroubleshootingApiData>(
-      `Troubleshooting/${wikiname}`,
-      'troubleshooting',
-      {
-         credentials: 'include',
+   let troubleshootingData: TroubleshootingApiData;
+   try {
+      troubleshootingData = await client.get<TroubleshootingApiData>(
+         `Troubleshooting/${wikiname}`,
+         'troubleshooting',
+         {
+            credentials: 'include',
+         }
+      );
+   } catch (e) {
+      // If fetch() doesn't error, e.message is the response's statusText
+      if (typeof e === 'object' && (e as Error)?.message === 'Not Found') {
```

Comment:
Let's turn this into a function like:

`function rethrowUnless404(e) { if (404) { return } else { throw e} }`

Then we can do `rethrowUnless404(e)` and use that in the several places here where we are using catch-alls.

Reply:
[44045fc](https://github.com/iFixit/react-commerce/pull/1705/commits/44045fc915e90cd2a1a70e0a05e089b5df396e44)